### PR TITLE
New excel range conditions

### PR DESCRIPTION
**Pull request by naterush**
# Description

Adds new start and end conditions. Also, refactors the frontend and backend code for _maximal flexibility_ on these going forward. You simply need to add the conditions in a few places, and you're good. No massive overhead of frontend development anymore.

# Testing

Use it. Adds 3 tests or so. **Feel free to add an ends with condition to test out how suitable the infrastructure is, if you want.**

# Documentation

We will have to update these docs.

### CODE REVIEW

**Comment by aarondr77** [comment]
The starts with condition is not working for me. 

See the following configuration with the following data.

[Tables.xlsx](https://github.com/mito-ds/monorepo/files/11217341/Tables.xlsx)


Screenshot 2023-04-12 at 9 30 47 PM

**Comment by aarondr77** [comment]
The top left corner contains is also not working

Screenshot 2023-04-12 at 9 32 48 PM

**Review by aarondr77 — COMMENTED** [review]
Once you fix up these not working conditions and do another thorough test, LGTM. 

I added a new end condition to test for X consecutive empty cells

File: mitosheet/mitosheet/tests/step_performers/import_steps/test_excel_range_import.py [hunk 1]
```diff
@@ -206,6 +207,14 @@
         [{'type': 'dynamic', 'start_condition': {'type': 'upper left corner value contains', 'value': 'er 10'}, 'end_condition': {'type': 'bottom left corner value contains', 'value': 'abc'}, 'column_end_condition': {'type': 'first empty cell'}, 'df_name': 'dataframe_1'}],
         [TEST_DF_4.iloc[0:1]],
     ),  
+    # Tests bottom left corner consecutive emtpy cells
```
Comment (naterush) [inline mitosheet/mitosheet/tests/step_performers/import_steps/test_excel_range_import.py:210]: Add a test of the opposite condition

**Review by naterush — COMMENTED** [review]
Cleanup

File: mitosheet/mitosheet/code_chunks/step_performers/import_steps/excel_range_import_code_chunk.py [hunk 2]
```diff
@@ -5,22 +5,105 @@
...
+        bottom_left_value_contains = end_condition['value']
+    else:
+        bottom_left_value_contains = None
+
+    if column_end_condition['type'] == EXCEL_RANGE_COLUMN_END_CONDITION_NUM_COLUMNS:
+        num_columns = column_end_condition['value']
+    else:
+        num_columns = None
+
+    all_params = {
```
[the remark this answers is not in the document]
Comment (naterush) [reply mitosheet/mitosheet/code_chunks/step_performers/import_steps/excel_range_import_code_chunk.py:100]: We'd just need to update the type in this case. Since the type is used in one place only, we really won't benefit from this change - we'd have to remember to type it correctly exactly once in either case.
